ml-agents/mlagents/trainers/trajectory.py

```diff
@@ -59,11 +74,97 @@ def from_buffer_next(batch: AgentBuffer, num_obs: int) -> List[np.array]:
         return result
 
 
+class GroupObsUtil:
+    @staticmethod
+    def get_name_at(index: int) -> AgentBufferKey:
+        """
+        returns the name of the observation given the index of the observation
+        """
+        return ObservationKeyPrefix.GROUP_OBSERVATION, index
+
+    @staticmethod
+    def get_name_at_next(index: int) -> AgentBufferKey:
+        """
+        returns the name of the next team observation given the index of the observation
+        """
+        return ObservationKeyPrefix.NEXT_GROUP_OBSERVATION, index
+
+    @staticmethod
```

Comment:
I think there is some duplicate code with agent_action.py

Reply:
Moved this code to AgentBufferField